vue-components/src/components/QuantityInput.vue

```diff
@@ -0,0 +1,221 @@
+			default: null,
+			validator: ( prop: string | null ): boolean => {
+				return [ 'number', 'unit', 'both', null ].includes( prop );
+			},
+		},
+		/**
+		 * visual label for the entire component, linked to the number input
+		 */
+		label: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * not visible, only accessible for screen readers
+		 */
+		unitLookupLabel: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * Placeholder for the number Input
+		 */
+		numberInputPlaceholder: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * This is the plain value in the number input. Note that it is a string
+		 */
+		numberInputValue: {
+			type: String,
+			default: null,
+		},
+		/**
+		 * Placeholder for the unit Lookup
+		 */
+		unitLookupPlaceholder: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * MenuItems for the unit Lookup
+		 */
+		unitLookupMenuItems: {
+			type: Array as PropType<MenuItem[]>,
+			required: true,
+		},
+		/**
+		 * Analogous to the `searchInput` prop on the Lookup component. Can be used with the .sync modifier
+		 */
+		unitLookupSearchInput: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * Analogous to the `value` prop on the Lookup component.
+		 * The selected menu item, can be of type `MenuItem` or `null`.
+		 *
+		 * Can be used with the .sync modifier.
+		 */
+		unitLookupValue: {
+			type: Object as PropType<MenuItem | null>,
+			default: null,
+		},
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	setup( props: { error: ErrorProp } ) {
+		return {
+			feedbackType: computed( getFeedbackTypeFromProps( props ) ),
+		};
+	},
+	data() {
+		return {
+			// https://github.com/vuejs/vue/issues/5886
+			id: generateUid( 'wikit-Input' ),
+		};
+	},
+	methods: {
+		onNumberInput( number: number ): void {
+			/**
+			 * The plain text in the number input as string
+			 */
+			this.$emit( 'update:numberInputValue', number );
+		},
+		onUnitLookupSearchInput( search: string ): void {
+			/**
+			 * The search text for the unit Lookup
+			 */
+			this.$emit( 'update:unitLookupSearchInput', search );
+		},
+		onUnitLookupValue( value: null | MenuItem ): void {
+			/**
+			 * The selected menu item for the unit Lookup
+			 */
+			this.$emit( 'update:unitLookupValue', value );
+		},
+	},
+	name: 'QuantityInput',
+	components: {
+		Input, // TODO: add validation
+		LookupInput,
+		ValidationMessage,
+	},
+} );
+</script>
+
+<style lang="scss">
+.wikit-QuantityInput {
+	&__input-wrapper {
+		display: flex;
+		align-items: baseline;
+	}
+
+	&__number-input {
+		flex: 1 1 50%;
+
+		&.wikit-Input {
+			border-start-end-radius: 0;
+			border-end-end-radius: 0;
+
+			&:not(.wikit-Input--error) {
+				border-inline-end-color: transparent;
+			}
+		}
+	}
+
+	&__unit-lookup {
+		flex: 1 1 50%;
+
+		.wikit-Input {
+			border-start-start-radius: 0;
+			border-end-start-radius: 0;
+		}
+
+		.wikit-Lookup__label {
+			position: absolute;
+			width: 1px;
+			height: 1px;
```

Comment:
there are no tokens here ?

Reply:
There could be? But I don't get to understand what are they styling 🤔

Reply:
Ah, this was hiding the Label from the Lookup component so that it is still readable to screen readers. But we're no longer using the Lookup, but the LookupInput, so this is now unused code and can be removed.

Reply:
great